src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java

```diff
@@ -604,30 +604,30 @@ public ResultSet getColumns(
             throws SQLException {
         MongoJsonSchema botSchema =
                 createBottomSchema(
-                        new Pair<>(TABLE_CAT, BSON_STRING.getBsonName()),
-                        new Pair<>(TABLE_SCHEM, BSON_STRING.getBsonName()),
-                        new Pair<>(TABLE_NAME, BSON_STRING.getBsonName()),
-                        new Pair<>(COLUMN_NAME, BSON_STRING.getBsonName()),
-                        new Pair<>(DATA_TYPE, BSON_INT.getBsonName()),
-                        new Pair<>(TYPE_NAME, BSON_STRING.getBsonName()),
-                        new Pair<>(COLUMN_SIZE, BSON_INT.getBsonName()),
-                        new Pair<>(BUFFER_LENGTH, BSON_INT.getBsonName()),
-                        new Pair<>(DECIMAL_DIGITS, BSON_INT.getBsonName()),
-                        new Pair<>(NUM_PREC_RADIX, BSON_INT.getBsonName()),
-                        new Pair<>(NULLABLE, BSON_INT.getBsonName()),
-                        new Pair<>(REMARKS, BSON_STRING.getBsonName()),
-                        new Pair<>(COLUMN_DEF, BSON_STRING.getBsonName()),
-                        new Pair<>(SQL_DATA_TYPE, BSON_INT.getBsonName()),
-                        new Pair<>(SQL_DATETIME_SUB, BSON_INT.getBsonName()),
-                        new Pair<>(CHAR_OCTET_LENGTH, BSON_INT.getBsonName()),
-                        new Pair<>(ORDINAL_POSITION, BSON_INT.getBsonName()),
-                        new Pair<>(IS_NULLABLE, BSON_STRING.getBsonName()),
-                        new Pair<>(SCOPE_CATALOG, BSON_STRING.getBsonName()),
-                        new Pair<>(SCOPE_SCHEMA, BSON_STRING.getBsonName()),
-                        new Pair<>(SCOPE_TABLE, BSON_STRING.getBsonName()),
-                        new Pair<>(SOURCE_DATA_TYPE, BSON_INT.getBsonName()),
-                        new Pair<>(IS_AUTOINCREMENT, BSON_STRING.getBsonName()),
-                        new Pair<>(IS_GENERATEDCOLUMN, BSON_STRING.getBsonName()));
+                        new MongoJsonSchema.ScalarProperties(TABLE_CAT, BSON_STRING),
```

Comment:
Looks like setting of nullability was missed for these properties in getColumns

Reply:
I did indeed, sorry for that.
I fixed that now, but since we can't test it until SQL-668 is merged, I think it's better wait and update the integration tests when SQL-668 is done. I'll create a JIRA to track that. Not only is_nullable but all metadata validation must be added to the integration tests.